### PR TITLE
Use Guice binding CatalogName in connectors

### PR DESCRIPTION
**Pull request by mx123**
## Description
Use Guice binding CatalogName in connectors since it provides within Guice connectors context.

### CODE REVIEW

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java [hunk 1]
```diff
@@ -69,8 +60,11 @@ public Connector create(String catalogName, Map<String, String> requiredConfig,
                 binder -> binder.bind(TypeManager.class).toInstance(context.getTypeManager()),
                 binder -> binder.bind(NodeManager.class).toInstance(context.getNodeManager()),
                 binder -> binder.bind(VersionEmbedder.class).toInstance(context.getVersionEmbedder()),
```
Comment (kokosing) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java:62]: place `binder -> binder.bind(CatalogName.class).toInstance(new CatalogName(catalogName))` here.

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java [hunk 2]
```diff
@@ -34,23 +35,13 @@
...
-     * @deprecated Prefer {@link JdbcConnectorFactory#JdbcConnectorFactory(String, Module)} instead.
-     * Notice that {@link io.trino.plugin.base.CatalogName} is available in guice context.
-     */
-    @Deprecated
-    public JdbcConnectorFactory(String name, JdbcModuleProvider moduleProvider)
     {
         checkArgument(!isNullOrEmpty(name), "name is null or empty");
         this.name = name;
-        this.moduleProvider = requireNonNull(moduleProvider, "moduleProvider is null");
+        this.module = module;
```
Comment (kokosing) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java:44]: use `requireNonNull`, please do the same for `name`.

**Review by kokosing — APPROVED** [review]

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java (hunk 1 above)
Comment (mx123) [reply plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java:62]: done

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java [hunk 3]
```diff
@@ -63,14 +54,16 @@ public String getName()
...
+        requireNonNull(module, "module is null");
         checkSpiVersion(context, this);
 
         Bootstrap app = new Bootstrap(
                 binder -> binder.bind(TypeManager.class).toInstance(context.getTypeManager()),
                 binder -> binder.bind(NodeManager.class).toInstance(context.getNodeManager()),
                 binder -> binder.bind(VersionEmbedder.class).toInstance(context.getVersionEmbedder()),
-                new JdbcModule(catalogName),
-                moduleProvider.getModule(catalogName));
+                binder -> binder.bind(CatalogName.class).toInstance(new CatalogName(catalogName)),
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcConnectorFactory.java:64]: This line and the lines around it strongly suggest we should consider moving the CatalogName into ConnectorContext. (Just from a consistency perspective - I don't know if it's technically possible or not though).

No change requested.

**Review by hashhar — APPROVED** [review]
I wonder what would change with dynamic catalogs being introduced.

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**